docs/CONTRIBUTING.md

```diff
@@ -1,21 +1,271 @@
+
+  For example, here are the first few lines of the @counter-style feature, which defines a list of BCD keys:
+
+  ```yaml
+  name: "@counter-style"
+  description: The `@counter-style` CSS at-rule defines custom counter styles for list items. For example, you can use a sequence of specific symbols instead of numbers for an ordered list.
+  spec: https://drafts.csswg.org/css-counter-styles-3/
+  caniuse: css-at-counter-style
+  compat_features:
+    - css.at-rules.counter-style
+    - css.at-rules.counter-style.additive-symbols
+    - css.at-rules.counter-style.fallback
+    - css.at-rules.counter-style.negative
+   ```
+
+  To identify the BCD keys that you need, check the browser compatibility tables displayed at the end of the MDN pages that document your new feature, and then search the BCD source code for the names that appear in the table.
+
+  You can also use the `npm run traverse | grep <keyword>` command in BCD to list all BCD keys that match a certain keyword.
+
+#### Generate and check the dist file
+
+To generate your feature's dist file, once your feature is associated with one or more BCD keys:
+
+1. Run the command `npm run dist`.
+
+1. Check that a new file, named after your feature ID, and with the `.yml.dist` file extension exists, and open the file.
+
+1. Check that the data in the dist file is correct. In particular, verify that the data is consistent with what developers would find when searching for the same feature on caniuse.com or MDN.
+
+   * Check the `baseline` status. If the feature isn't yet supported in all browsers of the core browsers set, it should be `false`. If the feature is supported on all browsers, it should either be `low` or `high`, depending on how long has passed.
+   * Check the `baseline_low_date` and `baseline_high_date` fields.
+   * Check that the browser versions listed under `support` match what caniuse.com and MDN document too.
+
+#### Fix data discrepancies in your feature's dist file
+
+The most likely reason for the data in your new feature's dist file to be incorrect is that your feature is associated to too many BCD keys.
+
+BCD keys describe all constituents of a feature, whether they are vital to the feature, or later additions to it. For example, developers have been able to use the Web Audio API for many years even if the AudioWorklet or OfflineAudioContext APIs were added later, and only recently became baseline.
+
+It's important to ensure that the BCD keys that your feature is associated with matches how most developers use the feature.
+
+To fix data discrepancies in your dist file, open the dist file and, under `compat_features`, review each individual section.
+
+Each section corresponds to a group of BCD keys that have the same baseline and support status.
+
+_TODO: continue documenting this section (consider splitting BCD keys into different features (later addition), or using just silencing some BCD keys by using `compat_from`)._
```

Comment:
This is the last part that I haven't yet documented. And it also happens to be the most tricky. I could use help figuring out how far to go. I feel like a lot of the times, this will be resolved by discussions in PRs, so I'm not sure how far I should go in trying to document each and every case.

Reply:
Yeah, I feel like this might be simplified a bit for now, until we have a better handle on common cases. I might simplify this whole section as having three key parts: too old, too new, and too wrong.

1. If the feature's overall status is too old (e.g., it's a new feature being reported as long-established), then you might have missing compat keys. Check that you've covering the complete feature by looking for missing interfaces, CSS property values, and so on.
1. If the feature's overall status is too young (e.g., a long-established feature is reported as newly available), then you might have later additions holding the feature back unfairly. Consider finding a small number of entry points into the feature that are representative of the whole (e.g., the `api.fetch` key can stand in for support for Fetch in general), then using those entry points in the `compute_from` field.
1. If the feature's overall status _and_ individual keys are yielding incorrect results, the underlying data may have caveats or errors. Check `mdn/browser-compat-data`. Important caveats include features behind prefixes or flags, or partial implementations. Errors include missing support (e.g., a feature shown as supported in a desktop browser but not mobile) or wrong version numbers.

Reply:
My approach is to look for the feature's entrypoint in the dist file. If that entrypoint doesn't have the "⬇️ Same status as overall feature ⬇️" comment I use `compute_from` to make it so. Whether the feature was too old or too new is something you find out this way, at least I don't do a different thing depending on whether I think it's too old or too new initially.